### ISSUE TITLE
Errors in `runtime.exs` not cleared

### ISSUE THREAD

**Issue by objectuser** [lexical-lsp/lexical#402]
Using VSCode, latest on `main` (`eeaa74f`), Elixir 1.15.4, Erlang 26.0.2, MacOS

If I introduce an error in `config/runtime.exs`, after fixing the error, the error is not cleared in the editor.

I think this is the only file that exhibits this behavior for me. It doesn't happen in other config files and it doesn't happen in test files.

I don't see anything specific to the issue in the console, just the `warn` deprecation log.

**Comment by zachallaun** [comment]
Looks like this can be reproduced in any `.exs` file, not just `config/runtime.exs`.

`scratch.exs` in the root of `lexical`:

```elixir
import Lexical.Server.IEx.Helpers

lx = project()
start_project(lx)

code = """
defmodule Foo do
  def bar do
    w|
  end
end
"""

complete(lx, code)
|> IO.inspect()
```

As I was typing this out, I initially got a `variable "lx" is unused` warning, which did not go away after it was used. It appears that new diagnostics are also not added, e.g. a new unused variable does not cause a new warning.

**Comment by scottming** [comment]
@objectuser This issue is fixed by #439 

@zachallaun 

Your reproduction method is incorrect because you used `start_project(lx)`, which would actually be executed and cause confusion in the communication between the editor's server and the project node, rather than being a diagnostics issue.

<img width="588" alt="CleanShot 2023-10-21 at 10 14 23@2x" src="https://github.com/lexical-lsp/lexical/assets/12830256/96acb537-a25b-40c5-a773-ce2e937ba44c">
